/*  Copyright 2012 MaidSafe.net limited

    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
    licence you accepted on initial access to the Software (the "Licences").

    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
    available at: http://www.maidsafe.net/licenses

    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
    OF ANY KIND, either express or implied.

    See the Licences for the specific language governing permissions and limitations relating to
    use of the MaidSafe Software.                                                                 */

option optimize_for = LITE_RUNTIME;

package maidsafe.vault_manager.protobuf;

message VaultInfo {
  required bytes pmid = 1;
  required bytes anpmid = 2;
  required bytes vault_dir = 3;
  required bytes label = 4;
  optional uint64 max_disk_usage = 5;
  optional bytes owner_name = 6;
}

message VaultManagerConfig {
  required bytes AES256Key = 1;
  required bytes AES256IV = 2;
  repeated VaultInfo vault_info = 3;
  optional bytes vault_permissions = 4;
}
